app/controllers/spree/locale_controller.rb

```diff
@@ -10,6 +10,11 @@ def set
       new_locale = (params[:switch_to_locale] || params[:locale]).to_s
 
       if new_locale.present? && supported_locale?(new_locale)
+
+        if try_spree_current_user && try_spree_current_user.selected_locale != new_locale
+          try_spree_current_user.update(selected_locale: new_locale)
```

Comment:
We could use `update!` instead - I don't think any validation would fail, but if it does, we would like to know about it. `update` won't let us know about this unfortunately.

```suggestion
          try_spree_current_user.update!(selected_locale: new_locale)
```

Reply:
makes sense, I have changed it